### PR TITLE
Fixed: Empty groups are still shown

### LINKED ISSUE #9521: When deleting only file of this day in downloads there remains tag saying Today 0 item
### Description

When you delete only file of today in downloads folder, there reamins label with text ```Today 0 item```.

### Steps To Reproduce

1) go to downloads while not having any file today
2) create new file
3) delete this file

### Expected behavior

When you delete only file of this day, there should not be any label about today

### Files Version

Version: 2.3.0.0 OS Version: 10.0.22000.778

### Windows Version

Windows 11 Home 21H2 22000.778

### Relevant Assets/Logs

image

image

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- Closes #9521
- Related #9895

**Details of Changes**
- Fixed: when an element is removed from a group, if the group is empty, it is deleted
- Added null checks

**Validation**
- Built and ran the app

### CODE REVIEW

File: src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs
Comment (yaichenbaum) [inline src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:177]:
```suggestion
            if (GroupedCollection is null)
                return;
```
We try to not use brackets for single line statements

Comment (ferrariofilippo) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:177]: Ok, no problem. I'll change this even in your code styles, with your permission

Screenshot (35)

Comment (yaichenbaum) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs]: That would be great, thanks for noticing that!

**Review by yaichenbaum — DISMISSED** [review]
LGTM

I'd like to wait for another review before merging

**Comment by Josh-65** [comment]
Does this fix  #5011 also?

**Comment by ferrariofilippo** [comment]
> Does this fix #5011 also?

No it doesn't. Anyway I can try to work on that

**Comment by yaichenbaum** [comment]
@ferrariofilippo btw, you can check our new project board to see which tasks have all the requirements figued out. Anything in the `ready to build` column has been reviewed. If something is not yet in the `ready to build` column and you want to work on it, mention me on the issue and I'll look it over beforehand so that there are no surprises.

**Comment by ferrariofilippo** [comment]
> @ferrariofilippo btw, you can check our new project board to see which tasks have all the requirements figured out. Anything in the `ready to build` column has been reviewed. If something is not yet in the `ready to build` column and you want to work on it, mention me on the issue and I'll look it over beforehand so that there are no surprises.

Ok, no problem

File: src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs
Comment (d2dyno1) [inline src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:162]: When can `GroupedCollection` be null?

Comment (ferrariofilippo) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:162]: At line @ 97 `GroupedCollection` may be assigned `null`

Comment (d2dyno1) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:162]: Why though? 🤔

Comment (ferrariofilippo) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:162]: So I did a little bit of testing to understand when it might be null and it seems that when you swap from `Home` to a folder `GroupedCollection` is assigned null
Null

Comment (d2dyno1) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:162]: Thanks!

**Review by d2dyno1 — APPROVED** [review]

File: src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs
Comment (yaichenbaum) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:162]: Can you add a comment to the code? Would be helpful for other devs.

Comment (ferrariofilippo) [reply src/Files.App/Helpers/ItemListDisplayHelpers/BulkConcurrentObservableCollection.cs:162]: Ok

**Review by yaichenbaum — APPROVED** [review]
LGTM

**Pull request merged**
